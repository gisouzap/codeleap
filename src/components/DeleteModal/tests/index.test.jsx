import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import DeleteModel from '../index';

describe('DeleteModel', () => {
  it('should not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');

    const handleDelete = jest.fn();
    const handleClose = jest.fn();

    render(<DeleteModel onDelete={handleDelete} onClose={handleClose} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should call handleClose when close button is clicked ', () => {
    const handleDelete = jest.fn();
    const handleClose = jest.fn();

    render(<DeleteModel onDelete={handleDelete} onClose={handleClose} />);

    const closeBtn = screen.getByRole('button', { name: /cancel/i });

    fireEvent.click(closeBtn);

    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('should call handleDelete when delete button is clicked ', () => {
    const handleDelete = jest.fn();
    const handleClose = jest.fn();

    render(<DeleteModel onDelete={handleDelete} onClose={handleClose} />);

    const deleteBtn = screen.getByRole('button', { name: /delete/i });

    fireEvent.click(deleteBtn);

    expect(handleDelete).toHaveBeenCalledTimes(1);
  });

  it('should call handleClose when ESC key is pressed', () => {
    const handleDelete = jest.fn();
    const handleClose = jest.fn();

    const { container } = render(
      <DeleteModel onDelete={handleDelete} onClose={handleClose} />
    );

    fireEvent.keyDown(container.querySelector('.modal'), { keyCode: 27 });

    expect(handleClose).toHaveBeenCalled();
  });

  it('should not call handleClose when other key is pressed', () => {
    const handleDelete = jest.fn();
    const handleClose = jest.fn();

    const { container } = render(
      <DeleteModel onDelete={handleDelete} onClose={handleClose} />
    );

    fireEvent.keyDown(container.querySelector('.modal'), { keyCode: 25 });

    expect(handleClose).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const handleDelete = jest.fn();
    const handleClose = jest.fn();

    const { container } = render(
      <DeleteModel onDelete={handleDelete} onClose={handleClose} />
    );

    expect(container).toMatchSnapshot();
  });
});
