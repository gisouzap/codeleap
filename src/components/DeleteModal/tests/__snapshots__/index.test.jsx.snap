// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModel should render and match the snapshot 1`] = `
<div>
  <div
    class="modal"
  >
    <div
      class="card"
      style="width: 670px;"
    >
      <p
        class="title"
      >
        Are you sure you want to delete this item?
      </p>
      <div
        class="buttons"
      >
        <button
          class="button light"
          type="button"
        >
          Cancel
        </button>
        <button
          class="button danger"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
